Microsoft.DotNet.Interactive/LanguageService/LanguageServiceExtensions.cs

```diff
@@ -28,5 +30,39 @@ public static T ToLspObject<T>(this JObject obj)
         {
             return obj.ToObject<T>(_jsonSerializer);
         }
+
+        private const string DataUriPrefix = "data:";
+
+        public static string DecodeDocumentFromDataUri(this string dataUri)
```

Comment:
I'd recommend introducing types here to encapsulate validation and parsing logic.

Reply:
Done.